src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -316,90 +317,87 @@ private string GenerateId()
-                    && trimPosition <= 119) //overflow suffix length is 8 + 1 for #.
+                if (parentId[trimPosition - 1] == s_internalIdDelimiter || parentId[trimPosition - 1] == s_externalIdDelimiter)
                     break;
                 trimPosition--;
             }
 
             //ParentId is not valid Request-Id, let's generate proper one.
             if (trimPosition == 0)
-                return generateRootId();
+                return GenerateRootId();
 
             //generate overflow suffix
-            byte[] bytes = new byte[4];
-            s_random.Value.NextBytes(bytes);
-
-            return parentId.Substring(0, trimPosition) + 
-                s_overflowDelimiter +
-                BitConverter.ToUInt32(bytes, 0).ToString("x8");
+            byte[] bytes = Guid.NewGuid().ToByteArray();
+            return parentId.Substring(0, trimPosition) + BitConverter.ToUInt32(bytes, 12).ToString("x8") + s_overflowDelimiter;
         }
 
-        private string generateRootId()
+        private string GenerateRootId()
         {
-            if (s_uniqPrefix == null)
-            {
-                // Here we make an ID to represent the Process/AppDomain.   Ideally we use process ID but 
-                // it is unclear if we have that ID handy.   Currently we use low bits of high freq tick 
-                // as a unique random number (which is not bad, but loses randomness for startup scenarios).  
-                Interlocked.CompareExchange(ref s_uniqPrefix, generateUniquePrefix(), null);
-            }
-
 #if DEBUG
-            string ret = s_uniqPrefix + OperationName + s_idDelimiter + Interlocked.Increment(ref s_currentRootId);
+            string ret = s_uniqPrefix + "-" + OperationName + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + s_internalIdDelimiter;
 #else           // To keep things short, we drop the operation name 
-            string ret = s_uniqPrefix + s_idDelimiter + Interlocked.Increment(ref s_currentRootId);
+            string ret = s_uniqPrefix + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + s_internalIdDelimiter;
 #endif
             return ret;
         }
 
+        private string _rootId;
+
         // Used to generate an ID 
-        private int _currentChildId;            // A unique number for all children of this activity.  
-        private static int s_currentRootId;      // A unique number inside the appdomain.
-        private static string s_uniqPrefix;
+        // A unique number for all children of this activity.  
+        private int _currentChildId;
 
-        private string _rootId;
-        private static char s_idDelimiter = '.';
+        // A unique number inside the appdomain. 
+        private static long s_currentRootId;
+
+        private static string s_uniqPrefix;
+        private static char s_internalIdDelimiter = '.';
```

Comment:
As more of a stylistic comment, I find that naming literal character/strings to have dubious value unless you REALLY think that value will change AND that value is used multiple pleases (in probably different files).   Otherwise I find the code to be CLEARER if you just use the literals (otherwise I have to go hunting them down.   If we never change them that code is BETTER.  Even if we do change them if it is easy to do a search (that is the use of those variables are contained within a file (as in this case)), then even doing the update (which is rare) is pretty easy.  

In particular things that are part of a standard are very unlikely to change so using he literals is OK.  

But if you really like the symbolic names, I won't veto them.